crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -220,8 +224,53 @@ impl Timeline {
     /// object to look up items.
     pub async fn item_by_event_id(&self, event_id: &EventId) -> Option<EventTimelineItem> {
         let items = self.inner.items().await;
-        let (_, item) = rfind_event_by_id(&items, event_id)?;
-        Some(item.to_owned())
+        if let Some((_, item)) = rfind_event_by_id(&items, event_id) {
+            return Some(item.to_owned());
+        }
+
+        let timeline_event = self.room().event(event_id).await.ok()?;
+        let raw_sync_event: Raw<AnySyncTimelineEvent> = timeline_event.event.cast();
+
+        let Ok(event) = raw_sync_event.deserialize_as::<AnySyncTimelineEvent>() else {
+            warn!("Unable to deserialize latest_event as an AnySyncTimelineEvent!");
+            return None;
+        };
+
+        let timestamp = event.origin_server_ts();
+        let sender = event.sender().to_owned();
+        let is_own = self.room().client().user_id().map(|uid| uid == sender).unwrap_or(false);
+
+        // If we don't (yet) know how to handle this type of message, return `None`
+        // here. If we do, convert it into a `TimelineItemContent`.
+        let item_content = TimelineItemContent::from_latest_event_content(event)?;
+
+        // let sender_profile = self
+        //     .room()
+        //     .profile_from_user_id(&sender)
+        //     .await
+        //     .map(TimelineDetails::Ready)
+        //     .unwrap_or(TimelineDetails::Unavailable);
+
+        let event_kind = RemoteEventTimelineItem {
+            event_id: event_id.to_owned(),
+            reactions: IndexMap::new(),
+            read_receipts: IndexMap::new(),
+            is_own,
+            is_highlighted: false,
+            encryption_info: None,
+            original_json: Some(raw_sync_event),
+            latest_edit_json: None,
+            origin: RemoteEventOrigin::Sync,
+        }
+        .into();
+
+        Some(EventTimelineItem::new(
+            sender,
+            TimelineDetails::Unavailable,
+            timestamp,
+            item_content,
+            event_kind,
+        ))
```

Comment:
So this is an experimental solution that I tried to address the issue that I referenced in the meta issue @bnjbvr @Hywan 

Essentially this tries synthesises a timeline item if it is not found directly in the timeline, to address the issue of needing a timeline item for the edit/reply APIs, that would otherwise fail in case the timeline item has not been loaded/paginated yet.

Ofc the code is rough and I am probably doing something wrong (for example I am building the content as a `latest_event`)but I wanted to che check if this would work, since the only thing I can work with is the `event_id`

This makes replies work well! For edits instead what happens is that I am able to send the edit event, and the message appears as edited but its content does not result as changed at all, but I think is because I am not generating the original content properly. Any ideas on how to improve upon this, or if there is a better solution to address the issue?

Reply:
I think the proper way would be to change the API to send_reply or edit an event, so that instead of a `EventTimelineItem` it gets passed another parameter shape that is all the fields that are required to reply/edit the target.

Then, you introduce two new APIs:

- given an `EventTimelineItem`, build this edit/reply parameter from it. This would be used every time there's a corresponding timeline item already existing in the timeline
- given an `EventId`, fetch this data and gather it inside an edit/reply parameter. This woudl be used every time the timeline item couldn't be found from the item in the timeline.

(Alternatively: you could also trigger back-paginations until you find the event in the timeline?)

Reply:
Okay for now I have removed this code, and see if I can directly change the behaviour of the edit and the reply functions